From 7b025882207c6bac1fbe6430aaf1f4924c86b6e7 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@st.com>
Date: Fri, 1 May 2020 17:58:02 +0200
Subject: [PATCH 0217/1695] plat-stm32mp1: define platform flavors per DTS file

Identify platform flavors from targeted embedded DTS file.

There are several SoC variants (157A, 157C, 157D and 157F) and
several ST boards (DK1, DK2, ED1, EV1).
Some ST boards embed a 500MByte DDR other a 1GByte DDR.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 94dd8955b8d0f3ed58d1b03746421d51b8a17260 plat-stm32mp1: define platform flavors per DTS file]
Change-Id: I91d513b9866505a7977d9ce2543b402359434805
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 34 ++++++++++++++++++++++++-----
 1 file changed, 29 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index a7a48d5bf..5379ac645 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -1,23 +1,39 @@
 # 1GB and 512MB DDR targets do not locate secure DDR at the same place.
 flavor_dts_file-157A_DHCOR_AVENGER96 = stm32mp157a-dhcor-avenger96.dts
 flavor_dts_file-157A_DK1 = stm32mp157a-dk1.dts
+flavor_dts_file-157A_ED1 = stm32mp157a-ed1.dts
+flavor_dts_file-157A_EV1 = stm32mp157a-ev1.dts
 flavor_dts_file-157C_DHCOM_PDK2 = stm32mp157c-dhcom-pdk2.dts
 flavor_dts_file-157C_DK2 = stm32mp157c-dk2.dts
 flavor_dts_file-157C_ED1 = stm32mp157c-ed1.dts
 flavor_dts_file-157C_EV1 = stm32mp157c-ev1.dts
+flavor_dts_file-157D_DK1 = stm32mp157d-dk1.dts
+flavor_dts_file-157D_ED1 = stm32mp157d-ed1.dts
+flavor_dts_file-157D_EV1 = stm32mp157d-ev1.dts
+flavor_dts_file-157F_DK2 = stm32mp157f-dk2.dts
+flavor_dts_file-157F_ED1 = stm32mp157f-ed1.dts
+flavor_dts_file-157F_EV1 = stm32mp157f-ev1.dts
 
 flavor_dts_file-135F_DK = stm32mp135f-dk.dts
 
 flavorlist-cryp-512M = $(flavor_dts_file-157C_DK2) \
+		       $(flavor_dts_file-157F_DK2) \
 		       $(flavor_dts_file-135F_DK)
 
-flavorlist-no_cryp-512M = $(flavor_dts_file-157A_DK1)
+flavorlist-no_cryp-512M = $(flavor_dts_file-157A_DK1) \
+			  $(flavor_dts_file-157D_DK1)
 
 flavorlist-cryp-1G = $(flavor_dts_file-157C_DHCOM_PDK2) \
 		     $(flavor_dts_file-157C_ED1) \
-		     $(flavor_dts_file-157C_EV1)
+		     $(flavor_dts_file-157C_EV1) \
+		     $(flavor_dts_file-157F_ED1) \
+		     $(flavor_dts_file-157F_EV1)
 
-flavorlist-no_cryp-1G = $(flavor_dts_file-157A_DHCOR_AVENGER96)
+flavorlist-no_cryp-1G = $(flavor_dts_file-157A_ED1) \
+			$(flavor_dts_file-157A_EV1) \
+			$(flavor_dts_file-157A_DHCOR_AVENGER96) \
+			$(flavor_dts_file-157D_ED1) \
+			$(flavor_dts_file-157D_EV1)
 
 flavorlist-no_cryp = $(flavorlist-no_cryp-512M) \
 		  $(flavorlist-no_cryp-1G)
@@ -26,7 +42,7 @@ flavorlist-512M = $(flavorlist-cryp-512M) \
 		  $(flavorlist-no_cryp-512M)
 
 flavorlist-1G = $(flavorlist-cryp-1G) \
-		  $(flavorlist-no_cryp-1G)
+		$(flavorlist-no_cryp-1G)
 
 flavorlist-MP15-HUK-DT = $(flavor_dts_file-157A_DK1) \
 			 $(flavor_dts_file-157C_DK2) \
@@ -35,10 +51,18 @@ flavorlist-MP15-HUK-DT = $(flavor_dts_file-157A_DK1) \
 
 flavorlist-MP15 = $(flavor_dts_file-157A_DHCOR_AVENGER96) \
 		  $(flavor_dts_file-157A_DK1) \
+		  $(flavor_dts_file-157A_ED1) \
+		  $(flavor_dts_file-157A_EV1) \
 		  $(flavor_dts_file-157C_DHCOM_PDK2) \
 		  $(flavor_dts_file-157C_DK2) \
 		  $(flavor_dts_file-157C_ED1) \
-		  $(flavor_dts_file-157C_EV1)
+		  $(flavor_dts_file-157C_EV1) \
+		  $(flavor_dts_file-157D_DK1) \
+		  $(flavor_dts_file-157D_ED1) \
+		  $(flavor_dts_file-157D_EV1) \
+		  $(flavor_dts_file-157F_DK2) \
+		  $(flavor_dts_file-157F_ED1) \
+		  $(flavor_dts_file-157F_EV1)
 
 flavorlist-MP13 = $(flavor_dts_file-135F_DK)
 
-- 
2.47.2

